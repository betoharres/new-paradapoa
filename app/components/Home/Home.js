import React, {useState, useContext} from 'react'
import {Platform} from 'react-native'
import {shape, func} from 'prop-types'
import {SearchBar, ListItem} from 'react-native-elements'
import {Container} from './Home.styles'

import {DatabaseContext} from '~/app/database'

export default function Home({navigation}) {
  const [search, setSearch] = useState('')
  const [busList, setBusList] = useState([])
  const conn = useContext(DatabaseContext)

  async function onPressBus(bus) {
    navigation.navigate('Bus', {bus})
  }

  async function onChangeSearchField(text) {
    setSearch(text)
    const [result] = await conn.executeSql(
      `SELECT * FROM buses WHERE code LIKE '%${text}%' \
        OR name LIKE '%${text}%' LIMIT 30`
    )
    const searchedBusList = result.rows.raw()
    setBusList(searchedBusList)
  }

  return (
    <Container>
      <SearchBar
        showCancel
        value={search}
        placeholder="Pesquisar..."
        searchIcon={{name: 'search'}}
        clearIcon={{name: 'close'}}
        onChangeText={onChangeSearchField}
        platform={Platform.select({ios: 'ios', android: 'android'})}
      />
      {busList.map((bus) => (
        <ListItem
          key={bus.id}
          title={`${bus.code} - ${bus.name}`}
          onPress={() => onPressBus(bus)}
        />
      ))}
    </Container>
  )
}

Home.propTypes = {
  navigation: shape({
    navigate: func.isRequired,
  }).isRequired,
}
