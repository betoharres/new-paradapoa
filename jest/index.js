export function mockProps({props = {}, params = {}} = {}) {
  const {navigation} = props;
  const navigate = (navigation && navigation.navigate) || jest.fn();
  const goBack = (navigation && navigation.goBack) || jest.fn();
  const setParams = (navigation && navigation.setParams) || jest.fn();
  const mockNavigation = {navigation: {navigate, goBack, setParams}};
  return {...{route: {params}}, ...props, ...mockNavigation};
}

export * from '@testing-library/react-native';
