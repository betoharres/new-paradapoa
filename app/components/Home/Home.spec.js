import React from 'react';
import Home from './Home';
import {render, mockProps} from '~/jest';

it('renders correctly', () => {
  const props = mockProps();
  const wrapper = render(<Home {...props} />);
  expect(wrapper.toJSON()).toMatchSnapshot();
});
