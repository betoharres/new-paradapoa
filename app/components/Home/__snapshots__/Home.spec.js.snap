// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "paddingBottom": 8,
            "paddingTop": 8,
          }
        }
      >
        <View
          style={
            Object {
              "paddingHorizontal": 0,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 0,
                "borderColor": "#86939e",
                "flexDirection": "row",
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "marginLeft": 8,
                  "marginVertical": 4,
                  "paddingRight": 4,
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "rgba(0, 0, 0, 0.54)",
                            "fontSize": 25,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "MaterialIcons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      î¢¶
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <TextInput
              allowFontScaling={true}
              editable={true}
              onBlur={[Function]}
              onCancel={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Pesquisar..."
              platform="android"
              rejectResponderTermination={true}
              showCancel={true}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "marginLeft": 24,
                  "marginRight": 8,
                  "minHeight": 40,
                }
              }
              testID="searchInput"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "marginRight": 8,
                  "marginVertical": 4,
                  "paddingRight": 4,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              />
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#ff190c",
                "fontSize": 12,
                "height": 0,
                "margin": 0,
                "padding": 0,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
